### PR TITLE
Switch from scoped_ptr to std::unique_ptr in common code

### PR DESCRIPTION
**Pull request by alexanderbobrovnik**
Test-On-Device: true

b/291356560

### CODE REVIEW

**Review by kaidokert — APPROVED** [review]
LGTM as long as the test pass on all devices. Please don't submit before the on-device tests are green both on Github and internally.

**Review by amurovanyi — APPROVED** [review]

**Pull request merged**
